### PR TITLE
add data source enterprise project support

### LINKED ISSUE #587: Feature request: enterprise project data source
Hi there,

Now, some resources have support `enterprise_project_id`, but users prefers to use the name of enterprise project.
Could some one add a new data source named `huaweicloud_enterprise_project`, thanks.

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccEnterpriseProjectDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccEnterpriseProjectDataSource_basic -timeout 360m
=== RUN   TestAccEnterpriseProjectDataSource_basic
--- PASS: TestAccEnterpriseProjectDataSource_basic (10.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       11.001s

### CODE REVIEW

**Comment by Jason-Zhang9309** [comment]
fixes #587

File: huaweicloud/data_source_enterprise_project.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package huaweicloud
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/huaweicloud/golangsdk/openstack/eps/v1/enterpriseprojects"
+)
+
+func dataSourceEnterpriseProject() *schema.Resource {
```
Comment (niuzhenguo) [inline huaweicloud/data_source_enterprise_project.go]: use DataSourceEnterpriseProject to make it exportable.

File: huaweicloud/provider.go [hunk 2]
```diff
@@ -207,6 +207,7 @@ func Provider() terraform.ResourceProvider {
 			"huaweicloud_dms_az":                      dataSourceDmsAZV1(),
 			"huaweicloud_dms_product":                 dataSourceDmsProductV1(),
 			"huaweicloud_dms_maintainwindow":          dataSourceDmsMaintainWindowV1(),
+			"huaweicloud_enterprise_project":          dataSourceEnterpriseProject(),
```
Comment (niuzhenguo) [inline huaweicloud/provider.go]: ditto.

File: huaweicloud/data_source_enterprise_project.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/data_source_enterprise_project.go]: fixed

File: huaweicloud/provider.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/provider.go]: fixed

File: docs/data-sources/eps.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+---
+subcategory: "Enterprise Project Management Service (EPS)"
+---
+
+# huaweicloud\_Enterprise\_project
+
+Use this data source to get a enterprise project from HuaweiCloud
```
Comment (ShiChangkuo) [inline docs/data-sources/eps.md]: an enterprise project

File: docs/data-sources/eps.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Example Usage
+
+```hcl
+data "huaweicloud_enterprise_project" "test" {
+	name = "test"
```
Comment (ShiChangkuo) [inline docs/data-sources/eps.md]: two spaces not tab

File: huaweicloud/data_source_enterprise_project_test.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+			return fmt.Errorf("enterprise project data source ID not set ")
+		}
+
+		return nil
+	}
+}
+
+const testAccEnterpriseProjectDataSource_basic = `
+data "huaweicloud_enterprise_project" "test" {
+	name = "default"
```
Comment (ShiChangkuo) [inline huaweicloud/data_source_enterprise_project_test.go]: ditto

File: docs/data-sources/eps.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+---
+subcategory: "Enterprise Project Management Service (EPS)"
+---
+
+# huaweicloud\_Enterprise\_project
```
Comment (niuzhenguo) [inline docs/data-sources/eps.md]: s/Enterprise/enterprise

File: docs/data-sources/eps.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/eps.md]: fixed

File: docs/data-sources/eps.md (hunk 4 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/eps.md]: fixed

File: huaweicloud/data_source_enterprise_project_test.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/data_source_enterprise_project_test.go]: fixed

File: docs/data-sources/eps.md (hunk 6 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/eps.md]: fixed

**Comment by niuzhenguo** [comment]
better to add a list of resources which support eps.

File: docs/data-sources/eps.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+data "huaweicloud_enterprise_project" "test" {
+  name = "test"
+}
+```
+
+## Resources Supported Currently:
+Service Name | Resource Name
+---- | ---
+VPC | huaweicloud_vpc<br>huaweicloud_vpc_eip<br>huaweicloud_vpc_bandwidth<br>huaweicloud_networking_secgroup
+ECS | huaweicloud_ecs_instance<br>huaweicloud_compute_instance
```
Comment (niuzhenguo) [inline docs/data-sources/eps.md]: huaweicloud_ecs_instance is deprecated, just ignore it here.

Comment (Jason-Zhang9309) [reply docs/data-sources/eps.md]: fixed

**Pull request merged**